### ISSUE TITLE
 $routes->resources new option for controller mapping

### ISSUE THREAD

**Issue by gugleboot** [cakephp/cakephp#10423]
* [x] enhancement

* CakePHP Version: 3.4.3.

### What you did
```
Router::scope('/api', function (RouteBuilder $routes) {
    $routes->extensions(['json']);
      $routes->resources('Gallery', [
            'controller' => 'Albums',           //controller mapping here ??
            'only' => ['create', 'index'],
            ]
        ]
    );
});
```

### What happened
error returned: Controller class Gallery could not be found. 

### What you expected to happen
similar to this behavior
`$routes->connect('/Gallery', ['controller' => 'Albums', 'action' => 'add']);`

**Comment by chrisShick** [comment]
@markstory If I wanted to make a pull request with this feature, which branch should I base it on?

**Comment by markstory** [comment]
@chrisShick Target the `3.next` branch please. That branch contains the 'next' release which is 3.5.

**Comment by markstory** [comment]
Closing as this was implemented in #10488
